src/main/kotlin/no/nav/familie/ef/sak/mapper/SøknadsskjemaMapper.kt

```diff
@@ -0,0 +1,311 @@
+                       erklæringOmSamlivsbrudd = tilDomene(it.erklæringOmSamlivsbrudd?.verdi),
+                       hvorMyeErDuSammenMedAnnenForelder = it.hvorMyeErDuSammenMedAnnenForelder?.verdi,
+                       beskrivSamværUtenBarn = it.beskrivSamværUtenBarn?.verdi)
+            }
+
+    private fun tilDomene(annenForelder: KontraktAnnenForelder?): AnnenForelder? =
+            annenForelder?.let {
+                AnnenForelder(ikkeOppgittAnnenForelderBegrunnelse = annenForelder.ikkeOppgittAnnenForelderBegrunnelse?.verdi,
+                              bosattNorge = annenForelder.bosattNorge?.verdi,
+                              land = annenForelder.land?.verdi,
+                              person = tilDomene(annenForelder.person?.verdi))
+            }
+
+    private fun tilDomene(aktivitet: KontraktAktivitet): Aktivitet =
+            Aktivitet(hvordanErArbeidssituasjonen = StringUtils.join(aktivitet.hvordanErArbeidssituasjonen.verdi, ";"),
+                      arbeidsforhold = tilArbeidsgivere(aktivitet.arbeidsforhold?.verdi),
+                      firmaer = tilFirmaer(aktivitet.firmaer?.verdi),
+                      virksomhet = tilDomene(aktivitet.virksomhet?.verdi),
+                      arbeidssøker = tilDomene(aktivitet.arbeidssøker?.verdi),
+                      underUtdanning = tilDomene(aktivitet.underUtdanning?.verdi),
+                      aksjeselskap = tilAksjeselskap(aktivitet.aksjeselskap?.verdi),
+                      erIArbeid = aktivitet.erIArbeid?.verdi,
+                      erIArbeidDokumentasjon = tilDomene(aktivitet.erIArbeidDokumentasjon?.verdi))
+
+    private fun tilFirmaer(list: List<KontraktSelvstendig>?): Set<Selvstendig>? =
+            list?.map {
+                Selvstendig(firmanavn = it.firmanavn.verdi,
+                            organisasjonsnummer = it.organisasjonsnummer.verdi,
+                            etableringsdato = it.etableringsdato.verdi,
+                            arbeidsmengde = it.arbeidsmengde?.verdi,
+                            hvordanSerArbeidsukenUt = it.hvordanSerArbeidsukenUt.verdi)
+            }?.toSet() ?: emptySet()
+
+    private fun tilAksjeselskap(list: List<KontraktAksjeselskap>?): Set<Aksjeselskap>? = list?.map {
+        Aksjeselskap(navn = it.navn.verdi,
+                     arbeidsmengde = it.arbeidsmengde?.verdi)
+    }?.toSet() ?: emptySet()
+
+    private fun tilDomene(underUtdanning: KontraktUnderUtdanning?): UnderUtdanning? =
+            underUtdanning?.let {
+                UnderUtdanning(skoleUtdanningssted = it.skoleUtdanningssted.verdi,
+                               linjeKursGrad = it.gjeldendeUtdanning!!.verdi.linjeKursGrad.verdi,
+                               fra = it.gjeldendeUtdanning!!.verdi.nårVarSkalDuVæreElevStudent.verdi.fra,
+                               til = it.gjeldendeUtdanning!!.verdi.nårVarSkalDuVæreElevStudent.verdi.til,
```

Comment:
Kan vi bruke elvis ?: med error("Mangler gjeldende utdanning for ....") ? Blir litt lettere å oppdage eventuelle feil?